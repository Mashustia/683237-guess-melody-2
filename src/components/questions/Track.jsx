import React, {Component} from 'react';
import PropTypes from 'prop-types';

export class Track extends Component {
  constructor(props) {
    super(props);

    this.state = {
      checked: false
    };

    this.onChangeValue = this.onChangeValue.bind(this);
  }

  onChangeValue() {
    this.setState({
      checked: !this.state.checked
    }, () => this.props.onChoose(this.props.id));
  }

  render() {
    const {id} = this.props;

    return (
      <div className='track'>
        <button className='track__button track__button--play' type='button'/>
        <div className='track__status'>
          <audio/>
        </div>
        <div className='game__answer'>
          <input
            className='game__input visually-hidden'
            type='checkbox'
            name='answer'
            checked={this.state.checked}
            id={`answer-${id}`}
            onChange={this.onChangeValue}
          />
          <label className='game__check' htmlFor={`answer-${id}`}>Отметить</label>
        </div>
      </div>
    );
  }
}

Track.propTypes = {
  onChoose: PropTypes.func.isRequired,
  id: PropTypes.number.isRequired
};

export default Track;
