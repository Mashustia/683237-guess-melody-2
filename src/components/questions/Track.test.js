import React from 'react';
import renderer from 'react-test-renderer';
import Track from './Track';

it(`Track component renders correctly`, () => {
  const component = renderer
    .create(<Track id={0} onChoose={jest.fn()} />)
    .toJSON();
  expect(component).toMatchSnapshot();
});
