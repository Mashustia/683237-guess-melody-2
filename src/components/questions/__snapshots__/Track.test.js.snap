// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Track component renders correctly 1`] = `
<div
  className="track"
>
  <button
    className="track__button track__button--play"
    type="button"
  />
  <div
    className="track__status"
  >
    <audio />
  </div>
  <div
    className="game__answer"
  >
    <input
      checked={false}
      className="game__input visually-hidden"
      id="answer-0"
      name="answer"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      className="game__check"
      htmlFor="answer-0"
    >
      Отметить
    </label>
  </div>
</div>
`;
